File: seal/src/main/java/com/GuoGuo/JuicyChat/model/VideoFilePlugin.java
package com.GuoGuo.JuicyChat.model;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.support.v4.app.Fragment;
import android.support.v4.content.ContextCompat;

import com.GuoGuo.JuicyChat.R;

import io.rong.imkit.RongExtension;
import io.rong.imkit.plugin.IPluginModule;

/**
 * Created by chenshuai12619 on 2017-12-29.
 */

public class VideoFilePlugin implements IPluginModule {
    
    public static VideoFilePlugin videoFilePlugin;
    
    public static VideoFilePlugin getInstance() {
        if (videoFilePlugin == null) {
            synchronized (VideoFilePlugin.class) {
                if (videoFilePlugin == null) {
                    videoFilePlugin = new VideoFilePlugin();
                }
            }
        }
        return videoFilePlugin;
    }
    
    @Override
    public Drawable obtainDrawable(Context context) {
        return ContextCompat.getDrawable(context, R.drawable.selector_video_file);
    }
    
    @Override
    public String obtainTitle(Context context) {
        return "视频文件";
    }
    
    @Override
    public void onClick(Fragment fragment, RongExtension rongExtension) {
    
    }
    
    @Override
    public void onActivityResult(int i, int i1, Intent intent) {
    
    }
}

File: seal/src/main/java/com/GuoGuo/JuicyChat/model/VideoFilePlugin.kt
package com.GuoGuo.JuicyChat.model

import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import android.support.v4.app.Fragment
import android.support.v4.content.ContextCompat

import com.GuoGuo.JuicyChat.R
import com.GuoGuo.JuicyChat.ui.activity.VideoSendActivity

import io.rong.imkit.RongExtension
import io.rong.imkit.plugin.IPluginModule

/**
 * Created by chenshuai12619 on 2017-12-29.
 */

object VideoFilePlugin : IPluginModule {

    override fun obtainDrawable(context: Context): Drawable {
        return ContextCompat.getDrawable(context, R.drawable.selector_video_file)
    }

    override fun obtainTitle(context: Context): String {
        return "视频文件"
    }

    override fun onClick(fragment: Fragment, rongExtension: RongExtension) {
        rongExtension.startActivityForPluginResult(
                Intent(fragment.context, VideoSendActivity::class.java)
                        .putExtra("targetId", rongExtension.targetId)
                        .putExtra("type", rongExtension.conversationType),
                1,
                this)
        rongExtension.collapseExtension()
    }

    override fun onActivityResult(i: Int, i1: Int, intent: Intent?) {

    }
}
